core/domain/exp_domain.py

```diff
@@ -848,20 +849,24 @@ def from_dict(cls, answer_group_dict):
         return cls(
             Outcome.from_dict(answer_group_dict['outcome']),
             [RuleSpec.from_dict(rs) for rs in answer_group_dict['rule_specs']],
+            answer_group_dict['training_data']
         )
 
-    def __init__(self, outcome, rule_specs):
+    def __init__(self, outcome, rule_specs, training_data):
         """Initializes a AnswerGroup domain object.
 
         Args:
             outcome: Outcome. The outcome corresponding to the answer group.
             rule_specs: list(RuleSpec). List of rule specifications.
+            training_data: list(answers). List of answers belonging to training
```

Comment:
answers is not a type; use the actual Python type / class of the variable being passed.

Reply:
Yes, but answers have different types depending on which interaction is being used. It can be dict/list/string anything.

Reply:
Ah ok. If it is a class then say list(ClassName). But if it could be dicts/lists/str/etc. then say list(*).

Reply:
Ah, ok. Will do.

Reply:
Done